config/routes.rb

```diff
@@ -298,11 +298,9 @@
     end
   end
 
-  resources :audits, :only => [:index] do
-    collection do
-      get 'auto_complete_search'
-    end
-  end
+  match '/audits/auto_complete_search' => 'audits#auto_complete_search', :via => [:get]
+  resources :audits, :only => [:index], constraints: ->(req) { req.format == :json }
+  match '/audits' => 'react#index', :via => [:get]
```

Comment:
Doesn't this override what was defined in the line before?

Why is the `AuditsController` still needed when you move `/audits` to the `ReactController`?

Reply:
We need the index method because it manipulates the audits data for the audits table component

Reply:
Then let's keep the original implementation. You should either have

```ruby
resources :audits, :only => [:index], constraints: ->(req) { req.format == :json }
```
(defines `/index`)

```ruby
match '/audits' => 'react#index', :via => [:get]
```
(defines `/index` as well)

Reply:
```ruby
resources :audits, only: [] do
    collection do
      get 'index', constraints: ->(req) { req.format == :json }
      get 'auto_complete_search'
    end
  end
```

Reply:
this doesn't work for me. did you test it?

Reply:
Timo, the original implementation is actually correct:

1. get /audits -> react controller (initial page load)
2. get /audits.json -> load audits data (only via json) would be nice to replace it one day with something else (graphql?) but currently there is a lot of logic creating that data structure
3. get /audits/auto completer - required to generate search auto completer.

for 3,  I would be happy if we eventually migrate all of the controllers mixing to a new end point, something like
```
/search-completer?controller=audits&query=abc...
```

Reply:
> did you test it?

I did not, sorry for the confusion. Should have added that as a comment.

> Timo, the original implementation is actually correct:

Ok, this is super hard to understand. Can we write that in a cleaner way?

Reply:
Tests for this are also missing, aren't they?